Analytics/models/users.py

```diff
@@ -0,0 +1,96 @@
+'''
+Data model class for User credentials 
+'''
+
+import json
+from db import db
+from datetime import datetime
+from sqlalchemy.exc import IntegrityError
+import bcrypt
+
+class Users(db.Model):
+    __tablename__ = 'users'
+
+    id = db.Column(db.Integer, primary_key=True)
+    fullname = db.Column(db.String(400), nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    password = db.Column(db.String(400), nullable = False)
+    admin = db.Column(db.Boolean)
+    activated = db.Column(db.Boolean) # When a user is added (by the Admin page), their activated field is False. They need to access the /register endpoint to change this field to True
+    timestamp = db.Column(db.DateTime)
+
+    def __init__(self, fullname, email, password, admin, activated, timestamp=None):
+        self.fullname = fullname
+        self.email = email
+        self.password = password
+        self.admin = admin
+        self.activated = activated
+
+        if timestamp is None:
+            timestamp = datetime.utcnow()
+
+        self.timestamp = timestamp
+
+    def __repr__(self):
+        return 'Users email is: %s' % self.email
+
+    def __str__(self):
+        return self.json()
+
+    def json(self):
+        """ Returns a JSON representation of Users attributes """
+        return {
+            'id': self.id,
+            'fullname': self.firstname,
+            'email': self.email,
+            'admin': self.admin,
+            'activated': self.activated
+        }
+
+    def save(self):
+        """ Adds the current values of the Users fields to SQLAlchemy session but does not write to the database """
```

Comment:
All docstrings should be in the imperative mood. 
Think:
"""
Add the current Users fields to the SQLAlchemy session
"""